projects/knora-ui/src/lib/viewer/values/base-value.component.ts

```diff
@@ -1,8 +1,9 @@
-import {Input} from '@angular/core';
+import { Input, Directive } from '@angular/core';
 import {CreateValue, ReadValue, UpdateValue} from '@knora/api';
 import {AbstractControl, FormControl, FormGroup, ValidatorFn, Validators} from '@angular/forms';
 import {Subscription} from 'rxjs';
 
+@Directive()
```

Comment:
@flavens Why is this needed? It's just a class that component class extends.

Reply:
@tobiasschweizer it has been added during the migration
check https://angular.io/guide/migration-undecorated-classes

Reply:
Thanks for this link, I was unaware of this:

> The presence of the @Directive decorator causes Angular to generate extra code for the affected class. If that decorator includes no properties (metadata), the directive won't be matched to elements or instantiated directly, but other classes that extend the directive class will inherit this generated code. You can think of this as an "abstract" directive.

So I think adding the decorator for an abstract class (cannot be instantiated) is correct.

Reply:
hmm, but it's a component, not a directive ...

Reply:
related question: https://github.com/angular/angular/issues/35366

Reply:
@Component inherits from @Directive(), so what we have seems right.

https://angular.io/api/core/Component

A component is essentially a directive with a template